client/me/purchases/manage-purchase/index.jsx

```diff
@@ -46,22 +46,33 @@ import {
 	showCreditCardExpiringWarning,
 	showEditPaymentDetails
 } from 'lib/purchases';
-import { getPurchase, goToList, isDataLoading, recordPageView } from '../utils';
+import { getPurchase, getSelectedSite, goToList, isDataLoading, recordPageView } from '../utils';
 import { isDomainProduct, isGoogleApps, isPlan, isSiteRedirect, isTheme } from 'lib/products-values';
 
 const ManagePurchase = React.createClass( {
 	propTypes: {
 		cart: React.PropTypes.object.isRequired,
 		selectedPurchase: React.PropTypes.object.isRequired,
-		selectedSite: React.PropTypes.object,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
 		destinationType: React.PropTypes.string
 	},
 
 	componentWillMount() {
+		if ( ! this.isDataValid() ) {
+			return page.redirect( paths.list() );
```

Comment:
This line made me do a double take for two reasons:
1. since [`componentWillMount`](https://facebook.github.io/react/docs/component-specs.html#mounting-componentwillmount) doesn't expect a return value I would expect to see this split into two lines.
```js
page.redirect( paths.list() );
return;
```
2. I admit I don't know fully whats going on around and outside of this component but redirecting after the component has mounted because of missing props seems like this could benefit from some rethinking. I think the redirect should happen in the controller.